component/src/main/java/io/siddhi/extension/store/rdbms/metrics/RDBMSMetrics.java

```diff
@@ -0,0 +1,292 @@
+import org.wso2.carbon.metrics.core.Counter;
+import org.wso2.carbon.metrics.core.Level;
+import org.wso2.carbon.si.metrics.core.internal.MetricsDataHolder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Class which holds the RDBMS metrics.
+ */
+public class RDBMSMetrics {
+    private static final Map<RDBMSDatabase, Long> RDBMS_LAST_RECEIVED_TIME_MAP = new HashMap<>();
+    private static final Map<RDBMSDatabase, RDBMSStatus> RDBMS_STATUS_MAP = new HashMap<>();
+    private static final Map<String, Boolean> RDBMS_STATUS_SERVICE_STARTED_MAP = new ConcurrentHashMap<>();
+    private static final Logger log = Logger.getLogger(RDBMSMetrics.class);
+    private final String siddhiAppName;
+    private final String tableName;
+    private final RDBMSDatabase rdbmsDatabase;
+    private final Stopwatch processingTime = Stopwatch.createUnstarted();
+    private String shortenJdbcUrl;
+    private String databaseName;
+    private String dbType;
+    private long lastInsertTime;
+    private long lastUpdateTime;
+    private long lastDeleteTime;
+    private boolean isInitialised;
+    private long lastChangeTime;
+
+
+    public RDBMSMetrics(String siddhiAppName, String url, String tableName) {
+        this.siddhiAppName = siddhiAppName;
+        this.tableName = tableName;
+        this.rdbmsDatabase = new RDBMSDatabase(siddhiAppName, url + ":" + tableName);
+        MetricsDataHolder.getInstance().getMetricService()
+                .counter(String.format("io.siddhi.SiddhiApps.%s.Siddhi.Store.RDBMS", siddhiAppName), Level.INFO).inc();
+        RDBMS_STATUS_SERVICE_STARTED_MAP.putIfAbsent(siddhiAppName, false);
+    }
+
+    public void updateTableStatus(ExecutorService executorService, String siddhiAppName) {
+        if (!RDBMS_STATUS_SERVICE_STARTED_MAP.get(siddhiAppName)) {
+            RDBMS_STATUS_SERVICE_STARTED_MAP.replace(siddhiAppName, true);
+            executorService.execute(() -> {
+                while (RDBMS_STATUS_SERVICE_STARTED_MAP.get(siddhiAppName)) {
+                    if (!RDBMS_STATUS_MAP.isEmpty()) {
+                        RDBMS_LAST_RECEIVED_TIME_MAP.forEach((rdbmsDatabase, lastReceivedTime) -> {
+                            if (rdbmsDatabase.siddhiAppName.equals(siddhiAppName)) {
+                                long idleTime = System.currentTimeMillis() - lastReceivedTime;
+                                if (idleTime / 1000 > 8) {
+                                    RDBMS_STATUS_MAP.replace(rdbmsDatabase, RDBMSStatus.IDLE);
+                                }
+                            }
+                        });
+                    }
+                    try {
+                        Thread.sleep(500);
+                    } catch (InterruptedException e) {
+                        log.error(siddhiAppName + ": Error while updating the tables status.");
```

Comment:
Add the "e" as well {(siddhiAppName + ": Error while updating the tables status.",e)}

Reply:
check other places as well.